src/main/scala/chisel3/util/BitPat.scala

```diff
@@ -171,4 +171,66 @@ sealed class BitPat(val value: BigInt, val mask: BigInt, width: Int) extends Sou
   }.mkString
 
   override def toString = s"BitPat($rawString)"
+
+  /**
+    * If a BitPat overlaps with another BitPat, i.e., exists x : UInt that (x === this) && (x === rhs)
+    *
+    * (lhs overlap rhs) iff bitwise: both care (lhs.mask & rhs.mask) => both equal (lhs.value == rhs.value)
+    */
+  def overlap(rhs: BitPat): Boolean = ((mask & rhs.mask) & (value ^ rhs.value)) == 0
```

Comment:
```suggestion
  def overlaps(rhs: BitPat): Boolean = ((mask & rhs.mask) & (value ^ rhs.value)) == 0
```

I'm not 100% sure on this but "overlaps" seems more consistent with the Scala naming terminology where such methods are usually singular tense (seemingly US English over British English where a Set would be a singular noun). `overlaps` is not a method defined on Scala Sets, but it might be useful in the ScalaDoc to mention that this is the inverse of `disjoint`.